Tools/autotest/pysim/util.py

```diff
@@ -137,9 +137,14 @@ def build_SITL(build_target, j=None, debug=False, board='sitl', clean=True, conf
     return True
 
 
-def build_examples(board, j=None, debug=False, clean=False):
+def build_examples(board, j=None, debug=False, clean=False, configure=True, math_check_indexes=False, extra_configure_args=[]):
     # first configure
-    waf_configure(board, j=j, debug=debug)
+    if configure:
+        waf_configure(board,
+                      j=j,
+                      debug=debug,
+                      math_check_indexes=math_check_indexes,
+                      extra_args=extra_configure_args)
```

Comment:
I wonder if we can do better here by keeping the arguments to configure and build as separate, opaque, objects.

ATM if we add something to the build options we'd have to go to each of the functions here and add that extra option in and then make sure we passed it into `waf_configure`.  If instead we just passed the dict in we could `waf_configure(board, **configure_opts)` and not have (e.g.) `build_tests` have to know about what we were passing to configure.